Algorithms/is_unique.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+# CtCI 6th Edition Problem 1.1
+# Implement an algorithm to determine if a string has all unique characters.
+
+
+def is_unique(string):
+    chars = set(string)
+    for char in chars:
+        if string.count(char) > 1:
```

Comment:
I had thought a bit about this and, assuming I kept the set, I thought this would be O(2N).

As in the `Multi-Part Algorithms: Add vs. Multiply`  section (pdf page 55) it says `If your algorithm is in the form "do this, then, when you're all done, do that" then you add the runtimes.` So I thought it would be O(N) for the set and O(N) for the count, so O(N + N), or O(2N).

If this is the case then according to the `Drop the Constants` (pdf page 53), it says O(2N) its actually just O(N).

If I understand things correctly (and I would say there is high likelihood I do not), I believe the concise answer is also just O(N). As generating a set is O(N) and `len` is O(1).